pkg/controllers/state/cluster.go

```diff
@@ -59,8 +59,11 @@ type Cluster struct {
 
 	// consolidated is a dirty bit that indicates that the cluster hasn't
 	// changed since last consolidation and avoids recomputation.
-	consolidated   atomic.Bool
-	consolidatedAt atomic.Int64
+	consolidated atomic.Bool
+	// setConsolidatedCalledAt is a timestamp marking the last time that the consolidated
+	// dirty bit has been updated due to a SetConsolidated() call. This means that if SetConsolidated(false)
+	// is called multiple times over some timeframe, this value will keep extending as a SetConsolidated call is made
+	setConsolidatedCalledAt atomic.Int64
```

Comment:
One thought -- we could replace these with a single "lastConsolidated" timestamp. When consolidated, we set it to now. When dirty, we set to 0.

Reply:
Good callout. Updated.